codalab/worker_manager/worker_manager.py

```diff
@@ -75,7 +82,10 @@ def start_worker_job(self):
 
     def run_loop(self):
         while True:
-            self.run_one_iteration()
+            try:
+                self.run_one_iteration()
+            except (urllib.error.URLError, http.client.HTTPException, socket.error, NotFoundError):
```

Comment:
Add comment saying that this is to be robust against server disconnecting.

Also, is there a way to localize the try/except more or is this as far in you can push it?

Reply:
I think this is as far as it can reasonably go while still maintaining good coverage.